tests/test_gear_yaml2cfg.py

```diff
@@ -12,6 +21,17 @@
     )
 
 
+@pytest.fixture
+def default_yaml_files():
+    """Return list of defaults.yaml file within the haddock src directory."""
+    all_defaults_yaml: list[str] = []
+    for path, _, files in os.walk(haddock3_source_path):
+        for name in files:
+            if fnmatch(name, MODULE_DEFAULT_YAML):
+                all_defaults_yaml.append(f"{path}/{MODULE_DEFAULT_YAML}")
+    return all_defaults_yaml
```

Comment:
This, and the function that is using this fixture are in the wrong place. What you are trying to do here is to check wether any of the default parameters of any of the modules are duplicated. 

This is an integration test, move it there